### PR TITLE
Fix graphql websocket protocol

### LINKED ISSUE #293: Fix implementation details of the GraphQL over WebSocket Protocol
https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md

Particular Issues:
- *Connected Phase* not honored, connection init is sent on each subscription and never on queries or mutations via websocket
- `GQL_CONNECTION_ACK` is ignored
- `GQL_CONNECTION_TERMINATE` is not sent before closing the websocket connection

### PR DESCRIPTION
**Pull request by rose-a**
Fixes #293

### CODE REVIEW

File: src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs [hunk 1]
```diff
@@ -468,9 +456,38 @@ private async Task ConnectAsync(CancellationToken token)
                 Debug.WriteLine($"new incoming message stream {_incomingMessages.GetHashCode()} created");
 
                 _incomingMessagesConnection = new CompositeDisposable(maintenanceSubscription, connection);
+                
+                var initRequest = new GraphQLWebSocketRequest
+                {
+                    Type = GraphQLWebSocketMessageType.GQL_CONNECTION_INIT,
+                    Payload = new GraphQLRequest()
```
Comment (sungam3r) [inline src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs:463]: new GraphQLRequest() ? It seems like there could be any type of payload.

Comment (rose-a) [reply src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs:463]: yeah, that should be replaced by the payload from #294... I'll fix this as soon as the former gets merged...

Comment (rose-a) [reply src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs:463]: Just didn't want to mix them up

Comment (sungam3r) [reply src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs:463]: ok

**Pull request merged**